sdk/core/azure-core/tests/test_streaming.py

```diff
@@ -150,6 +151,18 @@ def test_decompress_compressed_header(http_request):
     decoded = content.decode('utf-8')
     assert decoded == "test"
 
+
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+def test_decompress_compressed_header_offline(port, http_request):
+    request = http_request(method="GET", url="http://localhost:{}/streams/decompress".format(port))
+    with RequestsTransport() as sender:
+        response = sender.send(request, stream=True)
+        response.raise_for_status()
+        data = response.stream_download(sender, decompress=False)
```

Comment:
For this test scenario, we have a compressed file and we want to pass `decompress=True` when stream downloading the response. This will signal to decompress the body based on the `Content-Encoding` header and then we'll successfully download the file contents (i.e. "test")